include/tarxx.h

```diff
@@ -436,41 +491,81 @@ namespace tarxx {
                          tar_type type = tar_type::unix_v7)
             : file_(filename, std::ios::out | std::ios::binary),
               file_name_(filename),
+              file_buffer_used_(0),
               callback_(nullptr), mode_(output_mode::file_output),
               compression_(compression),
               type_(type), stream_file_header_pos_(-1), stream_block_ {0}, stream_block_used_(0)
         {
 #    ifdef WITH_LZ4
             init_lz4();
+#    else
+            file_buffer_.reserve(file_buffer_default_size_);
 #    endif
         }
 
         explicit tarfile(callback_t&& callback,
                          compression_mode compression = compression_mode::none,
                          tar_type type = tar_type::unix_v7)
-            : file_(), file_name_(), callback_(std::move(callback)), mode_(output_mode::stream_output),
+            : file_(), file_name_(), file_buffer_used_(0), callback_(std::move(callback)), mode_(output_mode::stream_output),
               compression_(compression),
               type_(type), stream_file_header_pos_(-1), stream_block_ {0}, stream_block_used_(0)
         {
 #    ifdef WITH_LZ4
             init_lz4();
+#    else
+            file_buffer_.reserve(file_buffer_default_size_);
 #    endif
         }
 
 #else
         explicit tarfile(const std::string& filename, tar_type type = tar_type::unix_v7)
-            : file_(filename, std::ios::out | std::ios::binary), file_name_(filename), callback_(nullptr), mode_(output_mode::file_output), type_(type), stream_block_ {0}, stream_file_header_pos_(-1), stream_block_used_(0)
+            : file_(filename, std::ios::out | std::ios::binary), file_name_(filename), file_buffer_used_(0), callback_(nullptr), mode_(output_mode::file_output), type_(type), stream_block_ {0}, stream_file_header_pos_(-1), stream_block_used_(0)
         {
+            file_buffer_.reserve(file_buffer_default_size_);
         }
 
         explicit tarfile(callback_t&& callback,
                          tar_type type = tar_type::unix_v7)
-            : file_(), file_name_(), callback_(std::move(callback)), mode_(output_mode::stream_output),
+            : file_(), file_name_(), file_buffer_used_(0), callback_(std::move(callback)), mode_(output_mode::stream_output),
               type_(type), stream_file_header_pos_(-1), stream_block_ {0}, stream_block_used_(0)
         {
+            file_buffer_.reserve(file_buffer_default_size_);
         }
 #endif
 
+        // delete the copy constructor and copy assignment, as multiple instances to the same file are not supported
+        tarfile(const tarfile&) = delete;
+        tarfile& operator=(const tarfile& other) = delete;
+
+        tarfile(tarfile&& other) noexcept
+        {
+            *this = std::move(other);
+        }
+
+        tarfile& operator=(tarfile&& other) noexcept
```

Comment:
Hmm, this doesn't seem correct. Files will be closed after the move-assignment. This is semantically unexpected, I'd argue. How can we make this a "real" move?

Reply:
The operator has been deleted in the latest version. Moving the tarfile opened a can of worms with a ton of gotchas which I wanted to prevent by deleting the operator entirely.